### ISSUE TITLE
how I acces to Class name ?

### ISSUE THREAD

**Issue by nebiyebulan** [ultralytics/yolov5#5294]
## Additional context
![Screenshot from 2021-10-22 12-11-37](https://user-images.githubusercontent.com/62153842/138428101-01f739b2-cba7-4428-8ece-dd5913515c1d.png)

@glenn-jocher @timstokman  @farleylai 
Hello there. I used this model in license plate recognition system. It is necessary to read the plate from left to right. But I didn't know how to map to the label data. How can I sort these data correctly?
Thank you

**Comment by glenn-jocher** [comment]
@nebiyebulan if you want to sort left right then just sort rows by `xmin`:

```bash
results.pandas().xyxy[0]
#      xmin    ymin    xmax   ymax  confidence  class    name
# 0  749.50   43.50  1148.0  704.5    0.874023      0  person
# 1  433.50  433.50   517.5  714.5    0.687988     27     tie
# 2  114.75  195.75  1095.0  708.0    0.624512      0  person
# 3  986.00  304.00  1028.0  420.0    0.286865     27     tie
```

See **PyTorch Hub** tutorial for details:

## YOLOv5 Tutorials

* [Train Custom Data](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data)&nbsp; 🚀 RECOMMENDED
* [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/)&nbsp; ☘️ RECOMMENDED
* [Weights & Biases Logging](https://github.com/ultralytics/yolov5/issues/1289)&nbsp; 🌟 NEW
* [Supervisely Ecosystem](https://docs.ultralytics.com/yolov5/tutorials/model_export8)&nbsp; 🌟 NEW
* [Multi-GPU Training](https://docs.ultralytics.com/yolov5/tutorials/multi_gpu_training)
* [PyTorch Hub](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading)&nbsp; ⭐ NEW
* [TorchScript, ONNX, CoreML Export](https://docs.ultralytics.com/yolov5/tutorials/model_export) 🚀
* [Test-Time Augmentation (TTA)](https://docs.ultralytics.com/yolov5/tutorials/test_time_augmentation)
* [Model Ensembling](https://docs.ultralytics.com/yolov5/tutorials/model_ensembling)
* [Model Pruning/Sparsity](https://docs.ultralytics.com/yolov5/tutorials/model_pruning_and_sparsity)
* [Hyperparameter Evolution](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution)
* [Transfer Learning with Frozen Layers](https://docs.ultralytics.com/yolov5/tutorials/transfer_learning_with_frozen_layers)&nbsp; ⭐ NEW
* [TensorRT Deployment](https://github.com/wang-xinyu/tensorrtx)

**Comment by nebiyebulan** [comment]
hello, as you said, I have sorted by xmin. but even though I have 8 characters, 9 characters are coming. 0 is at the beginning of the list and next to the characters comes something like n1 n2 n3. can you please tell me?
![11](https://user-images.githubusercontent.com/62153842/138649061-e79086c9-bc3c-464f-b0ce-75815d92128f.png)
@glenn-jocher 
@timstokman 
@cgerum

**Comment by glenn-jocher** [comment]
@nebiyebulan class names are model attributes:

```python
# PyTorch Hub
import torch

# Model
model = torch.hub.load('ultralytics/yolov5', 'yolov5s')
model.names[0] = 'ABC'

# Images
img = 'https://ultralytics.com/images/bus.jpg'

# Inference
results = model(img)
results.print()
results.save()
```

![image](https://user-images.githubusercontent.com/26833433/138675934-27eccb0f-26e7-4668-8174-c292a926510d.png)

**Comment by payanrayan** [comment]
Hello

I used this code

dirnameYolo="best.pt"
from ultralytics import YOLO
model = YOLO(dirnameYolo)
class_list = model.model.names
#print(class_list)

image = cv2.imread("aa.jpg")

results = model.predict(image)
result = results[0]

xyxy = result.boxes.xyxy.numpy()
confidence = result.boxes.conf.numpy()
class_id = result.boxes.cls.numpy().astype(int)

class_names = [class_list[x] for x in class_id]

if len(class_names) != len(xyxy):
    print("Error: Mismatch between class_names and coordinates")
else:
    sorted_indices = sorted(range(len(xyxy)), key=lambda i: xyxy[i][0])  # مرتب‌سازی بر اساس مقدار اول (x_min)
    sorted_class_names = [class_names[i] for i in sorted_indices]
    sorted_xyxy = [xyxy[i] for i in sorted_indices]
    sorted_confidence = [confidence[i] for i in sorted_indices]
    
    for class_label, coordinate, conf in zip(sorted_class_names, sorted_xyxy, sorted_confidence):
        print(f"{class_label}", end='')

**Comment by payanrayan** [comment]
Hello

I used this code

dirnameYolo="best.pt"
from ultralytics import YOLO
model = YOLO(dirnameYolo)
class_list = model.model.names
#print(class_list)

image = cv2.imread("aa.jpg")

results = model.predict(image)
result = results[0]

xyxy = result.boxes.xyxy.numpy()
confidence = result.boxes.conf.numpy()
class_id = result.boxes.cls.numpy().astype(int)

class_names = [class_list[x] for x in class_id]

if len(class_names) != len(xyxy):
    print("Error: Mismatch between class_names and coordinates")
else:
    sorted_indices = sorted(range(len(xyxy)), key=lambda i: xyxy[i][0])  # Sort by first value (x_min)
    sorted_class_names = [class_names[i] for i in sorted_indices]
    sorted_xyxy = [xyxy[i] for i in sorted_indices]
    sorted_confidence = [confidence[i] for i in sorted_indices]
    
    for class_label, coordinate, conf in zip(sorted_class_names, sorted_xyxy, sorted_confidence):
        print(f"{class_label}", end='')

**Comment by glenn-jocher** [comment]
@payanrayan hello,

It seems you are trying to sort the class names and coordinates based on the x-coordinate (x_min). The code you have shared looks correct in terms of sorting the values.

If you are encountering an error where the length of the class names is not equal to the length of the coordinates, it might indicate a mismatch between the data. Please double-check that the data is correctly aligned and encoded.

If the issue persists, please provide more information or specific error messages so that we can assist you further.

Thank you!
